Jenkinsfile

```diff
@@ -48,17 +48,35 @@ pipeline {
                             // Apply the template object from JSON file
                             openshift.apply(readFile(params.APP_TEMPLATE_PATH))
 
+                            // Process the template and return the Map of the result
+                            def model = openshift.process(params.IMAGE_STREAM_NAME,
+                                    "-p",
+                                    "TAG=${env.TAG}",
+                                    "IMAGESTREAM_TAG=${params.IMAGE_STREAM_LATEST_TAG}",
+                                    "REGISTRY_PROJECT=${params.REGISTRY_PROJECT}",
+                                    "REGISTRY=${params.REGISTRY_URI}")
+
+                            /* If the project's secret exists it must be removed at least in
+                             * the case of using a database along with an application.
+                             * Issue: Since the template is being process and applied with each
+                             * job run the secret is also being updated since it contains generate
+                             * values from the template.
```

Comment:
This makes me wonder if we should take this one step further: do we need to process/apply the whole template on every run? The update to the other objects in the template might not be as harmful as the secret's, but isn't it unnecessary anyway?